src/plots/linePlot.ts

```diff
@@ -226,8 +226,36 @@ export class Line<X> extends XYPlot<X, number> {
     return this;
   }
 
+  /**
+   * Gets if collapseDenseLines is enabled
+   *
+   * When collapseDenseLines is enabled, vertical or nearly vertical line
+   * segments that have the same floored x coordinate will be bucketed then
+   * drawn, drastically reducing the render time of dense line plots like
+   * timeseries. Only applies to the high performance "canvas" drawer.
```

Comment:
If we extract out the data-transformation logic we can use this in normal SVG lines as well